ocean_lib/exceptions.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+
+class ContractNotFound(Exception):
+    """
+    Contract address is not found
+    in the factory events
+    """
+
+
+class DDOError(Exception):
+    """
+    Generic DDO Error
+    """
+
+
+class InsufficientBalance(Exception):
+    """
+    The token balance is not sufficient
+    """
+
+
+class AssetsError(Exception):
+    """
+    Generic Assets Error
+    """
+
+
+class BPoolError(Exception):
+    """
+    Generic Balancer Pool Error
+    """
+
+
+class ExchangeError(Exception):
+    """
+    Generic Ocean Exchange Error
+    """
+
+
+class TransactionReverted(Exception):
```

Comment:
Suggest renaming: 
-TransactionReverted -> VerifyTxFailed
-Generic transaction revert error -> Transaction failed verification; tx may have reverted

Why: "transaction reverted" does not sound like an error. And in the places where it's used, it's always about a "verify tx" call that comes back with negative results.